src/lib/operators/abstract_operator.hpp

```diff
@@ -216,20 +229,24 @@ class AbstractOperator : public std::enable_shared_from_this<AbstractOperator>,
   std::shared_ptr<const AbstractOperator> _left_input;
   std::shared_ptr<const AbstractOperator> _right_input;
 
-  bool _executed = false;
-  std::atomic<bool> _execution_started = false;
-
   // Is nullptr until the operator is executed
   std::shared_ptr<const Table> _output;
 
   // Weak pointer breaks cyclical dependency between operators and context
   std::optional<std::weak_ptr<TransactionContext>> _transaction_context;
 
+ private:
   // We track the number of consuming operators to automate the clearing of operator results.
-  std::atomic<int> _consumer_count = 0;
-
+  std::atomic_int32_t _consumer_count = 0;
   // Determines whether operator results can be cleared via clear_output().
   bool _never_clear_output = false;
+
+  enum class OperatorState : uint8_t { Created = 0, Running = 1, Executed = 2, Cleared = 3 };
```

Comment:
Any reason you are hard-coding `uint8_t` and the numbers here?

> If the first enumerator does not have an initializer, the associated value is zero. For any other enumerator whose definition does not have an initializer, the associated value is the value of the previous enumerator plus one.

Reply:
No, can be removed then.